### ISSUE TITLE
Code Completion/ Auto-complete functionality

### ISSUE THREAD

**Issue by aayushai** [aayushai/TranspilerX#10]
### Description
- Implement an autocomplete feature in the code editor to enhance user experience by providing suggestions based on user input.
- Consider integrating existing libraries for autocomplete (e.g., CodeMirror, Monaco Editor).
- Autocomplete suggestions appear after a specified number of characters are typed.
- Users can easily select and insert suggestions.
- Functionality works across multiple programming languages supported by the editor.

**Comment by Pratik00019** [comment]
Hello @aayushai ,

I would love to contribute to this project. Could you please assign this issue to me?

Thank you!

**Comment by aayushai** [comment]
Sure Go Ahead! @Pratik00019

**Comment by aayushai** [comment]
@Pratik00019 what's the status?

**Comment by Pratik00019** [comment]
@aayushai Working on it.

**Comment by Pratik00019** [comment]
Hello @aayushai,

Could you please clarify if you would like suggestions in a format similar to the one shown in the image?
If that's not the case, could you provide more guidance on the type of suggestions you're looking for? For example, when typing "print" a second time, I see that recommendations already appear, as shown in the image.
Thank you!
![image](https://github.com/user-attachments/assets/84b05bb1-9ae8-4a82-a637-2fd3217c0928)

**Comment by aayushai** [comment]
Hi @Pratik00019 ,
I know that monaco editor supports auto-completion and I have already enabled it also, but here i am talking about IntelliSense hinting.

Refer these documentation you will undrerstand how to implement it:

- [Stackoverflow](https://stackoverflow.com/questions/69780709/how-can-i-add-auto-completion-to-a-browser-based-editor-using-monaco)
- [hackerearth](https://help.hackerearth.com/hc/en-us/articles/900000796563-the-monaco-editor)
- [mono](https://mono.software/2017/04/11/custom-intellisense-with-monaco-editor/)
- [Github](https://github.com/microsoft/monaco-editor/issues/2755)

**Comment by Pratik00019** [comment]
hey @aayushai  Do you want like this ? And when I saw javascript there are already lints available? so should i do this linting only for python, as I am contributing for the first time ever
![image](https://github.com/user-attachments/assets/461d32e8-2639-4103-a920-fca82389be03)

**Comment by aayushai** [comment]
Yes @Pratik00019 for Javascript and Typescript linting is there, you have to do for rest of the languages.

**Comment by Pratik00019** [comment]
@aayushai As I have added for def in python, do you want like that ?
